voter-dapp/src/ActiveRequests.js

```diff
@@ -90,18 +90,30 @@ function ActiveRequests({ votingAccount, votingGateway }) {
 
   // Only display non-blacklisted price requests (uniquely identifier by identifier name and timestamp)
   const [showSpamRequests, setShowSpamRequests] = useState(false);
+  const [hasSpamRequests, setHasSpamRequests] = useState(false);
   const [pendingRequests, setPendingRequests] = useState([]);
   useEffect(() => {
+    setHasSpamRequests(false);
+
     if (allPendingRequests) {
-      const _pendingRequests = allPendingRequests.filter(req => {
-        if (showSpamRequests || !REQUEST_BLACKLIST[hexToUtf8(req.identifier)]) return true;
+      const _nonBlacklistedRequests = allPendingRequests.filter(req => {
+        if (!REQUEST_BLACKLIST[hexToUtf8(req.identifier)]) return true;
         else {
-          if (!REQUEST_BLACKLIST[hexToUtf8(req.identifier)].includes(req.time)) return true;
-          else return false;
+          if (!REQUEST_BLACKLIST[hexToUtf8(req.identifier)].includes(req.time)) {
+            return true;
+          } else return false;
```

Comment:
nit: this may not be better, but just wanted to get your thoughts on a slightly shorter implementation:
```
const blacklistedTimestamps = REQUEST_BLACKLIST[hexToUtf8(req.identifier)];
return blacklistedTimestamps && blacklistedTimestamps.includes(req.time);
```

Reply:
its more concise but I think less readable, IMO